### PR TITLE
Add project URLs into the package metadata for convenience links on Pypi

### LINKED ISSUE #5865: No convenience links on https://pypi.org/project/datalad
On [Pypi](https://pypi.org/project/datalad/), there are no links in the sidebar to the project homepage/documentation/bugtracker like in many other projects. I think this is due to missing optional package metadata.

### PR DESCRIPTION
**Pull request by adswa**
This PR adds optional package metadata in the form of ``project_urls``. I believe they would show up on pypi.org in the side bar, making it easier to find documentation, source, or bug tracker. Fixes  #5865

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -192,6 +192,11 @@
         requires['core'] + requires['downloaders'] +
         requires['publish'] + requires['metadata'],
     python_requires='>=3.6',
+    project_urls={'Homepage': 'https://www.datalad.org',
+                  'Developer docs': 'http://docs.datalad.org/en/stable',
+                  'User handbook': 'http://handbook.datalad.org',
```
Comment (yarikoptic) [inline setup.py:197]:
```suggestion
                  'Developer docs': 'https://docs.datalad.org/en/stable,
                  'User handbook': 'https://handbook.datalad.org',
```
stay "safe" for all? ;)

**Comment by yarikoptic** [comment]
@jwodder -- please share your wisdom on this PR -- will it result in desired effect on pypi?

**Comment by jwodder** [comment]
@yarikoptic Yes.

**Review by jwodder — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (adswa) [reply setup.py]: thanks for the suggestion! I applied it

File: setup.py [hunk 2]
```diff
@@ -192,6 +192,11 @@
         requires['core'] + requires['downloaders'] +
         requires['publish'] + requires['metadata'],
     python_requires='>=3.6',
+    project_urls={'Homepage': 'https://www.datalad.org',
+                  'Developer docs': 'https://docs.datalad.org/en/stable,
```
Comment (yarikoptic) [inline setup.py:196]: I broke it :-/
```suggestion
                  'Developer docs': 'https://docs.datalad.org/en/stable',
```

**Comment by yarikoptic** [comment]
uff, all green now, let's proceed!

**Pull request merged**